cmd/object-handlers.go

```diff
@@ -771,12 +900,43 @@ func (api objectAPIHandlers) PutObjectHandler(w http.ResponseWriter, r *http.Req
 		}
 	}
 
-	hashReader, err := hash.NewReader(reader, size, md5hex, sha256hex)
+	hashReader, err := hash.NewReader(reader, size, md5hex, sha256hex, size)
 	if err != nil {
 		writeErrorResponse(w, toAPIErrorCode(err), r.URL)
 		return
 	}
 
+	storageInfo := objectAPI.StorageInfo(context.Background())
+
+	// Disabling compression for encrypted enabled requests.
+	// Since compression before encryption weakens the ciphertext.
+	// May lead to CRIME/BREACH.
```

Comment:
This comment is a bit frivolous we should talk about vulnerability only, not use words like CRIME/BREACH. `Using compression and encryption together enables room for side channel attacks`

Reply:
To clarify (CRIME/BREACH) are the names of the (TLS) side-channel attacks compression attacks -  However agree that a more generic comment is better...